tools/run_tests/python_utils/bazel_report_helper.py

```diff
@@ -0,0 +1,252 @@
+        workspace_status_command = 'tools/remote_build/workspace_status_kokoro.bat'
+    else:
+        workspace_status_command = 'tools/remote_build/workspace_status_kokoro.sh'
+
+    # generate RC file with the bazel flags we want to use apply.
+    # Using an RC file solves problems with flag ordering in the wrapper.
+    # (e.g. some flags need to come after the build/test command)
+    with open(bazel_rc_filename, 'w') as f:
+        f.write('build --invocation_id="%s"\n' % invocation_id)
+        f.write('build --workspace_status_command="%s"\n' %
+                workspace_status_command)
+
+    # generate "failing" and "success" report
+    # the "failing" is named as "sponge_log.xml", which is the name picked up by sponge/resultstore
+    # so the failing report will be used by default (unless we later replace the report with
+    # one that says "success"). That way if something goes wrong before bazel is run,
+    # there will at least be a "failing" target that indicates that (we really don't want silent failures).
+    with open(failing_report_filename, 'w') as f:
+        f.write(
+            _generate_junit_report_string(report_suite_name,
+                                          invocation_id,
+                                          success=False))
+    with open(success_report_filename, 'w') as f:
+        f.write(
+            _generate_junit_report_string(report_suite_name,
+                                          invocation_id,
+                                          success=True))
+
+    # generate the bazel wrapper for linux/macos
+    with open(bazel_wrapper_filename, 'w') as f:
+        intro_lines = [
+            '#!/bin/bash',
+            'set -ex',
+            '',
+            'tools/bazel --bazelrc="%s" "$@" || FAILED=true' %
+            bazel_rc_filename,
+            '',
+        ]
+
+        if upload_results:
+            upload_results_lines = [
+                '# Sleep to let ResultStore finish writing results before querying',
+                'sleep 60',
+                'PYTHONHTTPSVERIFY=0 python3 ./tools/run_tests/python_utils/upload_rbe_results.py --invocation_id="%s"'
+                % invocation_id,
+                '',
+            ]
+        else:
+            upload_results_lines = []
+
+        outro_lines = [
+            'if [ "$FAILED" != "" ]',
+            'then',
+            '  exit 1',
+            'else',
+            '  # success: plant the pre-generated xml report that says "success"',
+            '  mv -f %s %s' %
+            (success_report_filename, failing_report_filename),
+            'fi',
+        ]
+
+        lines = [
+            line + '\n'
+            for line in intro_lines + upload_results_lines + outro_lines
+        ]
+        f.writelines(lines)
+    os.chmod(bazel_wrapper_filename, 0o775)  # make the unix wrapper executable
+
+    # generate bazel wrapper for windows
+    with open(bazel_wrapper_bat_filename, 'w') as f:
```

Comment:
I would recommend to split the two generation process into two functions.

Reply:
I'll leave as is as the function already generates multiple different files and not just the two wrappers.